src/Entity.php

```diff
@@ -160,6 +160,34 @@ public function toArray($map = [])
         return $arr;
     }
 
+    /**
+     * Clones an entity and all nested entities
+     * Useful if you need to operate on a copy of an entity
+     * and not worry about mutating anything in the original
+     * @return Entity
+     */
+    public function fresh()
```

Comment:
@phily245 made a good point about the naming of this, `copy` would make more sense here

Reply:
Also need documenting in the README or something

Reply:
Oh, that's true. I'm thinking the Eloquent usage of fresh is for a new instance of the model. Whereas this is a copy of the current entity?

Reply:
yes